pubsub/pulsar/pulsar.go

```diff
@@ -101,16 +108,23 @@ func (p *Pulsar) Init(metadata pubsub.Metadata) error {
 	if err != nil {
 		return err
 	}
-	client, err := pulsar.NewClient(pulsar.ClientOptions{
-		URL:                        fmt.Sprintf("pulsar://%s", m.Host),
+	var pulsarURL string = m.Host
+	if !strings.HasPrefix(m.Host, "http") {
+		pulsarURL = fmt.Sprintf("%s%s", pulsarPrefix, m.Host)
+	}
+	options := pulsar.ClientOptions{
+		URL:                        pulsarURL,
 		OperationTimeout:           30 * time.Second,
 		ConnectionTimeout:          30 * time.Second,
 		TLSAllowInsecureConnection: !m.EnableTLS,
-	})
+	}
+	if m.Token != "" {
+		options.Authentication = pulsar.NewAuthenticationToken(m.Token)
+	}
+	client, err := pulsar.NewClient(options)
 	if err != nil {
 		return fmt.Errorf("could not instantiate pulsar client: %v", err)
 	}
-	defer client.Close()
```

Comment:
Why remove this?

Reply:
Originally, the client connection was created, and then the publish or subscribe operation was performed through the client. As a result, the client was closed, and the message could not be published.